src/RestApi/StoreApi/Utilities/MoneyValue.php

```diff
@@ -0,0 +1,91 @@
+<?php
+/**
+ * Handles monetary values using integers by using the smallest unit of a currency.
+ *
+ * This could be replaced with https://github.com/moneyphp/money if we want to
+ * support currencies in the future. However at this stage we only need to use
+ * the store decimal setting (which is independent of currency) to format
+ * values for the API.
+ *
+ * @package WooCommerce/Blocks
+ */
+
+namespace Automattic\WooCommerce\Blocks\RestApi\StoreApi\Utilities;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * MoneyValue class.
+ */
+class MoneyValue {
```

Comment:
With this Currency Vo, you can then instantiate a money value object with it and have a clear contract that the `MoneyValue` is a representation of money for a given currency (and there is no ambiguity).

Reply:
I'd like to keep the `MoneyValue` naming here, but I don't think we should grow the scope to include Currency objects for major currencies as seen in other implementations such as MoneyPHP. The largest reason being there are settings in core (DP) our of our control which may not necessarily match the currency.

So with that in mind, a currency class which allows us to plug in values from Woo core will be ideal. That would make it flexible enough to accept true currency objects in the future without increasing the scope too much for this work.

Reply:
> but I don't think we should grow the scope to include Currency objects for major currencies as seen in other implementations such as MoneyPHP.

I don't think we should implement setters that allow for switching currencies yet.  But I do think we need to implement the interface for constructing MoneyValue _with_ a currency object otherwise I'm sorry, but it's not a Money value object.  Call it something else (Number?) but it's not money.

> So with that in mind, a currency class which allows us to plug in values from Woo core will be ideal.

Yes that's exactly what I'm suggesting we start with initially.  It'd be a Currency value object that is immutable.  Then you'd just pass that into the MoneyValue object on construct.  Since MoneyValue would be immutable, and we're not exposing a way to change the currency (yet), we'd have an interface that is future proof but avoid scope creep of supporting multi-currency right away.

Reply:
> But I do think we need to implement the interface for constructing MoneyValue with a currency object otherwise I'm sorry, but it's not a Money value object. Call it something else (Number?) but it's not money.

Not disputing that at all. I think we agree. I'm just saying, like in the MoneyPHP library, I don't want to create objects for each currency e.g. a USD class, a EUR class and so on.

Reply:
💯